### PR TITLE
Add all assets of an asset type to a project at once

### LINKED ISSUE #468: [FEATURE] Add All Assets of an Asset Type
### Is there an existing issue or pull request for this?

- [X] I have searched the existing issues

### Is your feature request related to a problem? Please describe it

Prevents one getting RSI when adding all assets of a type

### Describe the solution you'd like

On the Add Asset search page next to the shopping basket there could be an add all assets button to add all of that type to a project 

### Describe alternatives you've considered

_No response_

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by cherry-john**
By submitting a PR for this repository you accept the contributor license agreement. 

### Description

Adds a new button to the Asset shopping popup to add all of one type of asset to the selected project.

### References

Closes #468 

### Checklist

- [x] I have added documentation for new/changed functionality in this PR or in the documentation repo
- [x] All active GitHub checks for tests, formatting, and security are passing
- [x] The correct base branch is being used, if not `main`

### CODE REVIEW

File: html/admin/assets.twig [hunk 1]
```diff
@@ -299,7 +302,7 @@
                                                                                             {%  if thisasset.assignment and searchResults.PROJECT.ID != thisasset.assignment[0].projects_id %}
                                                                                                 <button title="Assigned to {{ thisasset.assignment[0]['projects_name'] }}" class="btn btn-sm btn-success" disabled><i class="fa fa-shopping-basket"></i></button>
                                                                                             {% endif %}
-                                                                                            <button title="Add to project" {% if thisasset.assignment %} style="display:none;"{% else %}{% set someassignable = true %}{% endif %} class="btn btn-sm btn-success addToBasketAssetButton" data-assetid="{{ thisasset['assets_id'] }}"><i class="fa fa-shopping-basket"></i></button>
```
Comment (Jbithell) [inline html/admin/assets.twig:302]: I've removed other references to this variable https://github.com/adam-rms/adam-rms/commit/f270b80917f0268ff216ce257057fd54223e2465 - could you double check what I've done?

**Review by Jbithell — COMMENTED** [review]
I'm not sure about this one because it's quite a nice feature that you can add assets without it reloading at the moment - this change reloads the page each time you add/remove all assets for a type which means reloading all the images. These buttons are also broken by adding/removing individual assets (they don't update their count of assets available).

I would suggest ignoring a "remove all assets" button as you can do that quickly/easily in the asset list for a project, and it probably does a bit more harm than good as a button.

For adding all assets, how about doing it as a jquery selector: select all the "add to project" buttons with that data-typeid and just click them? That way the logic carries. Appreciate that's a bigger network load but saves re-doing the logic. Then to show/hide the button you can do that in a jquery selector as well based on a count of how many buttons there are saying add to project

File: html/admin/assets.twig (hunk 1 above)
Comment (cherry-john) [reply html/admin/assets.twig:302]: LGTM!

**Comment by cherry-john** [comment]
> I'm not sure about this one because it's quite a nice feature that you can add assets without it reloading at the moment - this change reloads the page each time you add/remove all assets for a type which means reloading all the images. These buttons are also broken by adding/removing individual assets (they don't update their count of assets available).
> 
> I would suggest ignoring a "remove all assets" button as you can do that quickly/easily in the asset list for a project, and it probably does a bit more harm than good as a button.
> 
> For adding all assets, how about doing it as a jquery selector: select all the "add to project" buttons with that data-typeid and just click them? That way the logic carries. Appreciate that's a bigger network load but saves re-doing the logic. Then to show/hide the button you can do that in a jquery selector as well based on a count of how many buttons there are saying add to project

Yeah agreed, have made the change to be JQuery based so should be a bit nicer on the user and the CDN!

**Review by Jbithell — APPROVED** [review]

**Pull request merged**